h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java

```diff
@@ -75,21 +78,24 @@ public static DMatrix convertFrameToDMatrix(Key<DataInfo> dataInfoKey,
             chunks = VecUtils.getLocalChunkIds(f.anyVec());
         }
         final Vec weightVector = f.vec(weight);
-        int nRows = sumChunksLength(chunks, vec, weightVector);
+        final int[] nRowsByChunk = new int[chunks.length];
+        final long nRowsL = sumChunksLength(chunks, vec, weightVector, nRowsByChunk);
+        if (nRowsL > Integer.MAX_VALUE) {
+            throw new IllegalArgumentException("XGBoost currently doesn't support datasets with more than " +
+                    Integer.MAX_VALUE + " per node. " +
+                    "To train a XGBoost model on this dataset add more nodes to your H2O cluster and use distributed training.");
+        }
+        final int nRows = (int) nRowsL;
```

Comment:
Ok, this might be dangerous ? Shouldn't we check for overflow ?

Reply:
It checks on like 83 :)